### DISCUSSION TITLE
how to select only key-value values - ignore objects

### DISCUSSION THREAD

**Discussion by mihaiav** [TomWright/dasel#230] [Q&A]
I have a TOML document and I would like to select only the "global" key/value pairs (APP_ID, DOMAIN_MAPPING, VERSION_ID) and ignore the values within an object (e.g. production object, sandbox object etc).
 Is it possible to do that? 

Please note that  the global values`APP_ID`, `DOMAIN_MAPPING`, `PROXY_SERVER` are arbitrary and there could be more/less values so I can't select them by name (i.e. `.APP_ID`)

```

APP_ID="x-example.com"
DOMAIN_MAPPING="false"
VERSION_ID = "XYZ"

[production]
ENV="production" 
PROXY_SERVER = "TRUE"

[sandbox]
SUFFIX_HOST="example.eu"
ENV="sandbox" 
PROXY_SERVER="FALSE"
DB_DISK="DISABLED"
```

`dasel -f env.toml -p toml '.ignore-objects-selector'`

**Expected result:**

```
APP_ID="x-example.com"
DOMAIN_MAPPING="false"
VERSION_ID = "XYZ"

```

I've tried ` dasel -f env.toml -p toml '.(!=.*)'` but it doesn't return the desired result..

**Comment by TomWright** [comment]
You could use dynamic selectors and `type` selector to filter out any `map` properties: `.(.[@]!=map)`, although this will only return the values.

The issue here is that there is no way to move back up the node tree once you select some nodes. I think to achieve this we'll need a new selector to move back up the node tree that will work alongside the filtering selectors.
